source/image/ImageBuffer.cpp

```diff
@@ -219,9 +224,7 @@ namespace {
 			return false;
 		}
 
-		// MAYBE: Reading in lots of images in a 32-bit process gets really hairy using the standard approach due to
-		// contiguous memory layout requirements. Investigate using an iterative loading scheme for large images.
-		png_init_io(png, file);
+		png_set_read_fn(png, file.get(), ReadPNGInput);
```

Comment:
Why the `.get()`?

Reply:
It expects a raw pointer.

Reply:
ah because file is a shared_ptr, nevermind